drivers/serial/uart_stm32.c

```diff
@@ -1512,6 +1545,12 @@ static int uart_stm32_async_tx(const struct device *dev,
 		return -EBUSY;
 	}
 
+#ifdef CONFIG_DCACHE
+	if (!buf_in_nocache((uintptr_t)tx_data, buf_size)) {
+		LOG_ERR("Tx buffer should be placed in a nocache memory region");
```

Comment:
should execution continue ?

Reply:
Having a DMA buffer in a cached memory region does not always mean the execution won't be successful, it only tells us that there may be cache coherence issues absent proper cache coherence management, that's why I didn't want to abort the execution. But I'm not sure if this is the proper why to treat it.

Reply:
Having a DMA buffer in a cached memory region does not always mean the execution won't be successful, it only tells us that there may be cache coherence issues absent proper cache coherence management, that's why I didn't want to abort the execution. But I'm not sure if this is the proper why to handle it.